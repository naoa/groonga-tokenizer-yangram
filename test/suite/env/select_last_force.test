#@omit "ommited."
#export GRN_VGRAM_NOT_FORCE_PREFIX=1
#export GRN_VGRAM_ADD_ALL_EXPAND_TOKEN=1

register tokenizers/yangram

table_create vgram_words TABLE_HASH_KEY ShortText
load --table vgram_words
[
{"_key": "データ"},
{"_key": "デー"}
]

table_create Memos TABLE_NO_KEY
column_create Memos content COLUMN_SCALAR ShortText

table_create Terms TABLE_PAT_KEY ShortText \
  --default_tokenizer TokenYaVgramQuad \
  --normalizer NormalizerAuto
column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content

load --table Memos
[
{"content": "動画データスデータス"}
]

select Memos --match_columns content --query "動画データスデータス" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画データスデータ" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画データスデー" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画データスデ" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画データス" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画データ" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画デー" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画デ" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動画" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "動" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "デ" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "デー" --output_columns content,_score --match_escalation_threshold -1
select Memos --match_columns content --query "データ" --output_columns content,_score --match_escalation_threshold -1

