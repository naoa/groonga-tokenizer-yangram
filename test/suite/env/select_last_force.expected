register tokenizers/yangram
[[0,0.0,0.0],true]
table_create vgram_words TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
load --table vgram_words
[
{"_key": "データ"},
{"_key": "デー"}
]
[[0,0.0,0.0],2]
table_create Memos TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Memos content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Terms TABLE_PAT_KEY ShortText   --default_tokenizer TokenYaVgramQuad   --normalizer NormalizerAuto
[[0,0.0,0.0],true]
column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content
[[0,0.0,0.0],true]
load --table Memos
[
{"content": "動画データスデータス"}
]
[[0,0.0,0.0],1]
select Memos --match_columns content --query "動画データスデータス" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画データスデータ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画データスデー" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画データスデ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画データス" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画データ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画デー" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画デ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動画" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "動" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "デ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        2
      ]
    ]
  ]
]
select Memos --match_columns content --query "デー" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
select Memos --match_columns content --query "データ" --output_columns content,_score --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "content",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "動画データスデータス",
        1
      ]
    ]
  ]
]
